trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/BookmarkManagerService.java

```diff
@@ -0,0 +1,474 @@
+        synchronized (EXPERIMENT_BOOKMARKS) {
+            List<Bookmark> bookmarks = Lists.transform(new ArrayList<>(EXPERIMENT_BOOKMARKS.getOrDefault(expUUID, Collections.emptyMap()).values()), bookmark -> bookmark);
+            return Response.ok(bookmarks).build();
+        }
+    }
+
+    /**
+     * Get a specific bookmark from an experiment
+     *
+     * @param expUUID
+     *            UUID of the experiment
+     * @param bookmarkUUID
+     *            UUID of the bookmark to retrieve
+     * @return Response containing the bookmark
+     */
+    @GET
+    @Path("/{bookmarkUUID}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Get a specific bookmark from an experiment", responses = {
+            @ApiResponse(responseCode = "200", description = "Returns the bookmark", content = @Content(schema = @Schema(implementation = Bookmark.class))),
+            @ApiResponse(responseCode = "404", description = "Experiment or bookmark not found", content = @Content(schema = @Schema(implementation = String.class)))
+    })
+    public Response getBookmark(
+            @Parameter(description = EXP_UUID) @PathParam("expUUID") UUID expUUID,
+            @Parameter(description = "Bookmark UUID") @PathParam("bookmarkUUID") UUID bookmarkUUID) {
+
+        TmfExperiment experiment = ExperimentManagerService.getExperimentByUUID(expUUID);
+        if (experiment == null) {
+            return Response.status(Status.NOT_FOUND).entity(NO_SUCH_EXPERIMENT).build();
+        }
+
+        Map<UUID, Bookmark> bookmarks = EXPERIMENT_BOOKMARKS.get(expUUID);
+        if (bookmarks == null || !bookmarks.containsKey(bookmarkUUID)) {
+            return Response.status(Status.NOT_FOUND).entity(EndpointConstants.BOOKMARK_NOT_FOUND).build();
+        }
+
+        return Response.ok(bookmarks.get(bookmarkUUID)).build();
+    }
+
+    /**
+     * Create a new bookmark in an experiment
+     *
+     * @param expUUID
+     *            UUID of the experiment
+     * @param queryParameters
+     *            Parameters for creating the bookmark
+     * @return Response containing the created bookmark
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Create a new bookmark in an experiment", responses = {
+            @ApiResponse(responseCode = "200", description = "Bookmark created successfully", content = @Content(schema = @Schema(implementation = Bookmark.class))),
+            @ApiResponse(responseCode = "400", description = INVALID_PARAMETERS, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = NO_SUCH_EXPERIMENT, content = @Content(schema = @Schema(implementation = String.class)))
+    })
+    public Response createBookmark(
+            @Parameter(description = EXP_UUID) @PathParam("expUUID") UUID expUUID,
+            @RequestBody(required = true) QueryParameters queryParameters) {
```

Comment:
in the RequestBody you need to add a description and ```@Schema``` to specify the model implementation `BookmarkQueryParameters`. Otherwise the swagger doc generation will references `QueryParameter`, but it doesn't exist. You can also add an ```@ExampleObject``` to provide an example.